### PR TITLE
[Resource] Refactor ibm_pi_placement_group resource

### PR DESCRIPTION
**Pull request by ismirlia**
Test:

```
--- PASS: TestAccIBMPIPlacementGroupBasic (3601.37s)
PASS
```

### CODE REVIEW

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 1]
```diff
@@ -944,12 +951,76 @@ func isPIInstanceRefreshFunc(client *st.IBMPIInstanceClient, id, instanceReadySt
...
 }
 
+func isWaitForPIInstancePlacementGroupAdd(ctx context.Context, client *st.IBMPIPlacementGroupClient, pgID string, id string) (interface{}, error) {
+	log.Printf("Waiting for PIInstance Placement Group (%s) to be updated ", id)
+
+	queryTimeOut := activeTimeOut
+
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{"notdone"},
+		Target:     []string{"done"},
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_instance.go:961]: We can have them as `adding` and `added` for pending and target.

File: ibm/service/power/resource_ibm_pi_placement_group.go [hunk 2]
```diff
@@ -35,37 +34,37 @@ func ResourceIBMPIPlacementGroup() *schema.Resource {
...
+				Description:  "The name of the placement group.",
+				Required:     true,
 				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_PlacementGroupPolicy: {
+				Description:  "The value of the group's affinity policy. Valid values are 'affinity' and 'anti-affinity'.",
 				Required:     true,
+				Type:         schema.TypeString,
 				ValidateFunc: validate.ValidateAllowedStringValues([]string{"affinity", "anti-affinity"}),
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_placement_group.go:54]: I have added allowed value in the constant file for those two as:
```
// Allowed Values
	Affinity                  = "affinity"
	AntiAffinity              = "anti-affinity"
```

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 3]
```diff
@@ -944,12 +953,76 @@ func isPIInstanceRefreshFunc(client *st.IBMPIInstanceClient, id, instanceReadySt
 	}
 }
 
+func isWaitForPIInstancePlacementGroupAdd(ctx context.Context, client *st.IBMPIPlacementGroupClient, pgID string, id string) (interface{}, error) {
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_instance.go:956]: Could you add constant for the one you just added ? `adding`, `added`, 'deleting', 'deleted'

**Review by michaelkad — CHANGES_REQUESTED** [review]
I missed those earlier

File: ibm/service/power/resource_ibm_pi_instance.go (hunk 3 above)
Comment (ismirlia) [reply ibm/service/power/resource_ibm_pi_instance.go:956]: Updated

**Review by michaelkad — APPROVED** [review]

**Comment by ismirlia** [comment]
Opened in the main terraform repo: https://github.com/IBM-Cloud/terraform-provider-ibm/pull/5335 closing

**Pull request closed without merging**